### ISSUE TITLE
Running home-manager activation inside a nix build sandbox

### ISSUE THREAD

**Issue by expipiplus1** [nix-community/home-manager#1717]
It would be great to be able to run the activation script from inside a nix-build sandbox.

This would be useful for running tests for one's dotfiles as I do here https://github.com/expipiplus1/dotfiles/tree/db638e7873fc10ad5e9a990f0612d0acbdd9d4c1/tests

I currently have something like this working, but it feels a little hacky in places: https://github.com/expipiplus1/dotfiles/blob/db638e7873fc10ad5e9a990f0612d0acbdd9d4c1/tests/home-test.nix#L69-L91

Hopefully it would be possible to make the activation script 'just work' inside such an environment (after setting `HOME`).

**Comment by expipiplus1** [comment]
An alternative would be to just put the code I linked in a test to make
sure that the existing ability to do it in a hacky way doesn't regress.

**Comment by expipiplus1** [comment]
hmm, had a quick look but it seems a little tricky to fit this kind of meta test neatly into nmt. What are the options here?

**Comment by expipiplus1** [comment]
still important.

**Comment by expipiplus1** [comment]
Still important

**Comment by rycee** [comment]
Agreed, I've been wanting this for the HM test suite as well. Sooner or later I'll get it fixed 🙂

**Comment by hab25** [comment]
I believe this would be very useful in lowering the amount of HM bugs, as well as bugs in users' own configuration.

**Comment by rycee** [comment]
For what it's worth, HM already has integration tests. You can run them using

```
$ nix build --no-sandbox -L --reference-lock-file flake.lock ./tests#integration-test-all
```

in the project root.
